Bottom: 60f33ac7a1ddb5218e2d351820645baa95ff6868
Top:    8b2bba3e6db687b768022942436aac12062dad38
Author: Lawrence D'Anna <lawrence_danna@apple.com>
Date:   2019-10-18 18:11:40 -0700

Refresh of no-more-reset

---
diff --git a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp
index db6b3f35af5..c345c05c844 100644
--- a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp
+++ b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.cpp
@@ -1040,8 +1040,12 @@ bool PythonException::Matches(PyObject *exc) const {
 }
 
 const char read_exception_script[] = R"(
-from io import StringIO
+import sys
 from traceback import print_exception
+if sys.version_info.major < 3:
+  from StringIO import StringIO
+else:
+  from io import StringIO
 def read_exception(exc_type, exc_value, tb):
   f = StringIO()
   print_exception(exc_type, exc_value, tb, file=f)
@@ -1540,7 +1544,12 @@ python::runStringOneLine(CStringArg string, const PythonDictionary &globals,
     return exception();
   auto code_ref = Take<PythonObject>(code);
 
+#if PY_MAJOR_VERSION < 3
+  PyObject *result = PyEval_EvalCode((PyCodeObject*)code, globals.get(), locals.get());
+#else
   PyObject *result = PyEval_EvalCode(code, globals.get(), locals.get());
+#endif
+
   if (!result)
     return exception();
